### PR TITLE
OpenCVE throws an error when trying to enrich a CVE

### PR DESCRIPTION
**Pull request by kobymeir**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-31415

## Description
OpenCVE throws an error when trying to enrich a CVE

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/OpenCVE/Integrations/OpenCVE/test_data/CVE-2021-26418.json [hunk 1]
```diff
@@ -1 +1,132 @@
...
+                    "availabilityImpact": "NONE",
+                    "privilegesRequired": "LOW",
+                    "confidentialityImpact": "LOW"
+                },
+                "impactScore": 2.5,
+                "exploitabilityScore": 2.1
+            }
+        },
+        "publishedDate": "2021-05-11T19:15Z",
+        "configurations": [{
```
Comment (kobymeir) [inline Packs/OpenCVE/Integrations/OpenCVE/test_data/CVE-2021-26418.json:104]: prettified json, and changed it to the new API response format, the configuration from a dict to list

File: Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py [hunk 2]
```diff
@@ -480,7 +480,17 @@ def parse_cve(ocve: OpenCVE, cve: dict):
...
-    parsed_cve['fields']['nodes'] = cve.get('raw_nvd_data', {}).get('configurations', {}).get('nodes', {})
+    # Backwards compatibility, the configuration used to be a dict. Now it is a list.
+    configurations = cve.get('raw_nvd_data', {}).get('configurations')
+    nodes = []
+    if isinstance(configurations, dict):
+        nodes = configurations.get('nodes', [])
+    elif isinstance(configurations, list):
+        for config in configurations:
+            nodes.extend(config.get('nodes', []))
+    # Design decision: The nodes are aggregated into a single list, this means we won't support 'AND' Statements in CPE matching,
```
Comment (kobymeir) [inline Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:491]: The design decision was done with Dror

Comment (tkatzir) [reply Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:491]: What's `CPE`? Maybe explain it somewhere?

Also, is `raw_nvd_data.configurations` always present?
If you already touch this area, maybe test that they exist?

File: Packs/OpenCVE/Integrations/OpenCVE/test_data/CVE-2021-26418.json (hunk 1 above)
Comment (tkatzir) [reply Packs/OpenCVE/Integrations/OpenCVE/test_data/CVE-2021-26418.json:104]: Is it a SaaS service?
Can't some customers be on the previous API version?
In that case, you'll need to add a `server_version` param.

**Review by tkatzir — APPROVED** [review]
Good!
Approved with minor comments.

File: Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py (hunk 2 above)
Comment (kobymeir) [reply Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:491]: CPE is a known security term, Common Platform Enumeration.

and the configuration might not exist, that's why i'm doing a `get` on it.

File: Packs/OpenCVE/Integrations/OpenCVE/test_data/CVE-2021-26418.json (hunk 1 above)
Comment (kobymeir) [reply Packs/OpenCVE/Integrations/OpenCVE/test_data/CVE-2021-26418.json:104]: this is just for Unit testing, and i'm doing backward compatibility check in the code. this API is in beta mode, so it's subject to change. but it only has a single version.

**Pull request merged**